policies/module-types/commands/src/lib.rs

```diff
@@ -0,0 +1,332 @@
+#[serde(rename_all = "snake_case")]
+pub struct CommandsParameters {
+    /// Command to be executed
+    command: String,
+
+    /// Arguments to the command
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_string"
+    )]
+    args: Option<String>,
+
+    /// Controls the running mode of the command
+    #[serde(default)] // Default to false
+    run_in_audit_mode: bool,
+
+    /// Controls if the command is executed inside a shell
+    #[serde(default)] // Default to false
+    in_shell: bool,
+
+    /// Shell path (used only in shell mode)
+    #[serde(default = "default_shell_path")]
+    shell_path: String,
+
+    /// Directory from where to execute the command
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_string"
+    )]
+    chdir: Option<String>,
+
+    /// Timeout for command execution
+    #[serde(default = "default_timeout")]
+    timeout: String, // Default to 30 seconds
+
+    /// Input passed to the stdin of the executed command
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_string"
+    )]
+    stdin: Option<String>,
+
+    /// Controls the appending of a newline to the stdin input
+    #[serde(default = "default_as_true")]
+    stdin_add_newline: bool,
+
+    /// Compliant codes
+    #[serde(default)] // Default to ""
+    compliant_codes: String,
+
+    /// Repaired codes
+    #[serde(default = "default_repaired_codes")]
+    repaired_codes: String, // Default to "0"
+
+    /// File to store the output of the command
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_pathbuf"
+    )]
+    output_to_file: Option<PathBuf>,
+
+    // Controls the strip of the content inside the output file
+    #[serde(default)] // Default to false
+    strip_output: bool,
+
+    /// UID used by the executed command
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_string"
+    )]
+    uid: Option<String>,
+
+    /// GID used by the executed command
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_string"
+    )]
+    gid: Option<String>,
+
+    /// Umask used by the executed command
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_string"
+    )]
+    umask: Option<String>,
+
+    /// Environment variables used by the executed command
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_string"
+    )]
+    env_vars: Option<String>,
+
+    /// Controls output of diffs in the report
+    #[serde(default = "default_as_true")]
+    show_content: bool,
+}
+
+fn deserialize_option_pathbuf<'de, D>(deserializer: D) -> Result<Option<PathBuf>, D::Error>
```

Comment:
IMO these functions should be in the `impl Commands` as they are only used inside it.